From e5198367fa8fac686628dc5e67cbcad809b43f97 Mon Sep 17 00:00:00 2001
From: Dan Albert <danalbert@google.com>
Date: Tue, 12 Dec 2017 21:12:55 -0800
Subject: [PATCH 1/5] Make rehash(0) work with ubsan's
 unsigned-integer-overflow.

Reviewers: mclow.lists, EricWF

Subscribers: cfe-commits

Differential Revision: https://reviews.llvm.org/D40743
(cherry picked from commit 4e689750214209b1de74bbfd512eb28099f22caa)

Change-Id: I1651ca10f611de4d5af213fa8465658156d772f0
---
 include/__hash_table | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/__hash_table b/include/__hash_table
index dac4cfddb..4fcdb4566 100644
--- a/include/__hash_table
+++ b/include/__hash_table
@@ -1932,7 +1932,7 @@ template <class _Tp, class _Hash, class _Equal, class _Alloc>
 void
 __hash_table<_Tp, _Hash, _Equal, _Alloc>::rehash(size_type __n)
 {
-    if (__n == 1)
+    if (__n < 2)
         __n = 2;
     else if (__n & (__n - 1))
         __n = __next_prime(__n);
-- 
2.17.1

